src/repli1d/nn.py

```diff
@@ -1,18 +1,62 @@
+import numpy as np
 import pandas as pd
-
-from keras.models import Sequential
-from keras.layers.core import Dropout, Dense, Activation, Flatten , Reshape
+from keras import backend as K
+from keras.callbacks import (EarlyStopping, History, ModelCheckpoint,
+                             ReduceLROnPlateau)
 from keras.layers.convolutional import Conv2D, MaxPooling2D
+from keras.layers.core import Activation, Dense, Dropout, Flatten, Reshape
 from keras.layers.recurrent import LSTM
 from keras.layers.wrappers import Bidirectional
-
-from keras.callbacks import EarlyStopping, History
-from keras import backend as K
-from keras.callbacks import ModelCheckpoint, ReduceLROnPlateau
+from keras.models import Sequential, load_model
 
 from repli1d.analyse_RFD import nan_polate, smooth
-from keras.models import load_model
-import numpy as np
+
+
+def normal_seq(signal, output_path='../data/'):
+    """
+    normalization function that transforms each fature in range (0,1)
+    and outputs the minimum and maximum of features in a csv file in 
+    data folder inside the repository, suitable for future transformation
+    on new dataset in a trained
+    neural network.
+
+    Parameters
+    ----------
+    signal : numpy array or pandas dataframe
+    in the shape of (n_samples, n_features)
+    output_path : str
+
+    Returns
+    -------
+    transformed : numpy array
+        a normalized sequence or features in the range (0,1)
+    """
+    global max_element, min_element
+    max_element = []
+    min_element = []
+    transformed = []
+    if isinstance(signal, pd.DataFrame):
+        signal = signal.to_numpy()
+    elif isinstance(signal, list):
+        signal = np.array(signal)
+    if signal.ndim == 1:
+        max_element = max(signal)
+        min_element = min(signal)
```

Comment:
Usually min and max are not really robust in biology because you can have very large outsider point.
It is probably better to use 10% quantile and 90 %, except maybe for the initiation as it should be between 0 and one.

Reply:
if outliers are due to experimental error or noise, and they have no information about the output, it's correct. otherwise using min/max can be a better idea because it can cover more range of inputs(less need to extrapolation for NN).
In any case having the constants for transforming new data on the same scale as training data is a good idea, I can apply the normalization method based on your suggestion, please let me know if the outliers are noise.

Reply:
Outliers are typically experimental error, because it comes from sequencing data et as some sequence that are localised at different positon are mapped sometimes at a unique position, and this can create artificially very high peaks.
But I agree with you that it is better pratice to store this information.